h/views/admin_users.py

```diff
@@ -141,17 +141,29 @@ def delete_user(request, user):
     message.
     """
 
-    if models.Group.created_by(request.db, user).count() > 0:
-        raise UserDeletionError('Cannot delete user who is a group creator.')
-
-    user.groups = []
+    # Check whether this user has created any groups that others have annotated
+    # in.
+    group_ids = [g.pubid for g in user.groups]
+    other_user_group_anns = request.db.query(models.Annotation) \
+                                      .filter(models.Annotation.groupid.in_(group_ids),
```

Comment:
One strange thing that I believe SQLAlchemy does is that, if you run an `IN` query on an empty set of IDs, it’ll evaluate it as `SELECT […] FROM annotation WHERE annotation.groupid != annotation.groupid`, causing a sequence scan on the whole table. In this case, you’d need something to shortcut the logic and set it to zero.